library/pk.c

```diff
@@ -404,10 +393,8 @@ int mbedtls_pk_verify_restartable( mbedtls_pk_context *ctx,
                const unsigned char *sig, size_t sig_len,
                mbedtls_pk_restart_ctx *rs_ctx )
 {
-    PK_VALIDATE_RET( ctx != NULL );
-    PK_VALIDATE_RET( ( md_alg == MBEDTLS_MD_NONE && hash_len == 0 ) ||
-                     hash != NULL );
-    PK_VALIDATE_RET( sig != NULL );
+    if( ( md_alg != MBEDTLS_MD_NONE || hash_len != 0 ) && hash == NULL )
```

Comment:
I would expect two test cases for this, as there are two ways to get `MBEDTLS_ERR_PK_BAD_INPUT_DATA` returned (same for the other two functions that return `MBEDTLS_ERR_PK_BAD_INPUT_DATA`)

Reply:
added 3 new tests